File: app/src/main/java.backup/com/lumiyaviewer/lumiya/ui/common/StateAwareFragment.java
package com.lumiyaviewer.lumiya.ui.common;

import android.support.v4.app.Fragment;

public class StateAwareFragment extends Fragment {
    private boolean fragmentStarted = false;
    private boolean fragmentVisible = false;

    public boolean isFragmentStarted() {
        return this.fragmentStarted;
    }

    public boolean isFragmentVisible() {
        return this.fragmentVisible;
    }

    public void onPause() {
        this.fragmentVisible = false;
        super.onPause();
    }

    public void onResume() {
        super.onResume();
        this.fragmentVisible = true;
    }

    public void onStart() {
        super.onStart();
        this.fragmentStarted = true;
    }

    public void onStop() {
        this.fragmentStarted = false;
        super.onStop();
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/ui/common/StateAwareFragment.kt
package com.lumiyaviewer.lumiya.ui.common

import android.support.v4.app.Fragment

class StateAwareFragment : Fragment {
    private Boolean fragmentStarted = false
    private Boolean fragmentVisible = false

    Boolean isFragmentStarted() {
        return this.fragmentStarted
    }

    Boolean isFragmentVisible() {
        return this.fragmentVisible
    }

    Unit onPause() {
        this.fragmentVisible = false
        super.onPause()
    }

    Unit onResume() {
        super.onResume()
        this.fragmentVisible = true
    }

    Unit onStart() {
        super.onStart()
        this.fragmentStarted = true
    }

    Unit onStop() {
        this.fragmentStarted = false
        super.onStop()
    }
}
